core/serialization/deserializer/montage-reviver.js

```diff
@@ -189,12 +189,12 @@ var MontageReviver = exports.MontageReviver = Montage.specialize(/** @lends Mont
                 Object.defineProperty(element, "dataset", {
                     value: new Proxy(targetObject, {
                         set: function (target, propertyName, value) {
+                            target[propertyName] = value;
+                            originalDataset[propertyName] = value;
                             element.nativeSetAttribute('data-' +
                                 kebabCaseConverter.convert(propertyName),
```

Comment:
Might be better to do 'data-' += kebabCaseConverter.convert(propertyName), it should spare 1 string, and if that were frequent, it would be worth caching the conversion in a map propertyName -> 'data-' +kebabCaseConverter.convert(propertyName), rather than re-doing it every time